### PR TITLE
Removed polling and used wait to process the next schedule event.

### PR DESCRIPTION
**Pull request by cableramki**

### CODE REVIEW

File: src/schedule.c [hunk 1]
```diff
@@ -245,13 +246,19 @@ char* get_blocked_at_time( schedule_t *s, time_t unixtime )
          * as it's in the past.  Otherwise use the weekly schedule. */
         if( (w_prev->time < last_abs) && (last_abs <= weekly) ) {
             rv = __convert_event_to_string( s, abs_prev );
+            *next_unixtime = s->absolute->time;
         } else {
             rv = __convert_event_to_string( s, w_prev );
+            if( NULL != w_cur ) {
+                *next_unixtime = convert_weekly_time_to_unix(w_cur->time);
+            } else {
+                *next_unixtime = convert_weekly_time_to_unix(s->weekly->time);
```
Comment (schmidtw) [inline src/schedule.c]: Change this code to:
```
*next_unixtime = (7 * 24 * 3600) - weekly + s->weekly->time + unixtime;
```

File: src/time.h [hunk 2]
```diff
@@ -33,4 +33,15 @@
...
+ *  Does the conversion from weekly time, seconds since Sunday midnight
+ *  to unixtime.
+ *
+ *  @param weekly_time the relative/weekly time in seconds since Sunday midnight, 
+ *                     i.e. Saturday 23:59:59 + one second, to convert from
+ *
+ *  @return the time since Epoch in seconds
+ */
+time_t convert_weekly_time_to_unix( time_t weekly_time );
+
```
Comment (schmidtw) [inline src/time.h]: This isn't needed.

File: src/time.c [hunk 3]
```diff
@@ -58,6 +58,27 @@ time_t convert_unix_time_to_weekly(time_t unixtime)
     return seconds_since_sunday_midnght;
 }
 
+/* See time.h for details. */
+time_t convert_weekly_time_to_unix(time_t weekly_time)
```
Comment (schmidtw) [inline src/time.c]: just get rid of this entire function

File: tests/test_schedule.c [hunk 4]
```diff
@@ -107,6 +107,12 @@ time_t convert_unix_time_to_weekly( time_t unixtime )
     return unixtime - 1234000 + 11;
 }
 
+time_t convert_weekly_time_to_unix( time_t weekly_time )
```
Comment (schmidtw) [inline tests/test_schedule.c]: Get rid of this

File: tests/test_schedule.c [hunk 5]
```diff
@@ -170,27 +177,27 @@ void test_simple_case( void )
 
     print_schedule( s );
 
-    block = get_blocked_at_time( s, 1233999 );
+    block = get_blocked_at_time( s, 1233999, &next_unixtime );
```
Comment (schmidtw) [inline tests/test_schedule.c]: You should verify the `next_unixtime` value is correct.

File: src/scheduler.c [hunk 6]
```diff
@@ -118,17 +121,24 @@ void *scheduler_thread(void *args)
...
     call_firewall( firewall_cmd, NULL );
 
-    while (1) {
-        struct timespec tm;
+    while( true ) {
+        if( 0 == clock_gettime(CLOCK_REALTIME, &tm) ) {
+            unix_time = tm.tv_sec; // ignore tm.tv_nsec
+            break;
+        }
+        sleep(SLEEP_TIME);
```
[the remark this answers is not in the document]
Comment (cableramki) [reply src/scheduler.c]: If clock_gettime() should return error.

File: tests/test_schedule.c (hunk 5 above)
Comment (cableramki) [reply tests/test_schedule.c]: Done.

**Comment by cableramki** [comment]
Closing now, to resubmit with more cleanup and tests.

**Pull request closed without merging**